### ISSUE TITLE
 Disabling room boss's voteless kick command in rooms that have at some point been unpassworded

### ISSUE THREAD

**Issue by sprangg** [ZeroK-RTS/Zero-K-Infrastructure#2739]
Would be best if voteless kicks by room bosses were disabled in rooms that have at some point been unpassworded, because this is against the CoC in most cases where it happens.

If an unpassworded room becomes passworded again, the voteless kick command should stay disabled, to prevent circumventing this with temporary !setpassword.

**Comment by GoogleFrog** [comment]
No. Open rooms with kicks have many applications. It is against the CoC when the boss essentially prevents the kicked player from playing ZK by attracting everyone to their room.

**Comment by sprangg** [comment]
I'm curious about your straight up refusal, GoogleFrog. Could you give me one example where the room boss's instant kick ability is able to accomplish something in an open room that a votekick couldn't do just as well?

As far as I see it, open unpassworded rooms are invitations for anyone to be able to join and play. Sometimes these rooms become the biggest room, but no matter the room size, bosses shouldn't get the power to decide who plays and who doesn't - that's for the people to decide with !votekick.

**Comment by GoogleFrog** [comment]
Say you're trying to make a 2v2 room but one player is preventing the game from starting because nobody wants to play with them. But more generally, I don't think arguing for disabling host kick without arguing for removing the concept of non-password hosts is coherent. If you remove powers then there is essentially no reason to have public player hosted rooms.

I'm fine with people kicking people they don't want to play with from their hosted room, as long as the kick doesn't prevent them from playing ZK entirely. Player hosting with effective powers is a potential route away from one-room-culture, and I don't want to close it off. I expect abusive hosts to find other ways to be abusive so they will need moderator attention regardless.

**Comment by sprangg** [comment]
The issue with kicks is that the rules change too much based on the situation. For example, the 2v2 room that you mentioned where one player got a kick from could grow to become 5v5 and the main team room for the evening, while the 2v2 is being played. After the 2v2 ends, kicking the same player would now be punishable? It would be simpler and clearer for all parties involved if the kicks were just vote-based, even if the voting requires some extra hurdles.

I don't see unpassworded player-hosted rooms as meaningless even if the kicks were removed. They still allow for easier customization of the options and maps. In fact, if the kicks were the only selling point, I'd say just get rid of the whole hosting thing.

Abusive hosts may find other ways, but there's no other simple way to target specific players than kick.

**Comment by maackey** [comment]
IMO if you host a room you should have control over that room. Kicks, map changes, exiting, etc. Because you are the host. This is pretty simple, clear and easy to understand. Disabling kicks in player hosted rooms is introducing a rule change based on a specific situation. Ostensibly what you just tried to argue against. 

If you host a room and then are a complete prick to everyone else -- that is a separate issue... kicks aren't the punishable element. Especially when games get big, the chances of players not paying attention (and voting) grow -- which make it increasingly difficult to deal with troublesome players. If a moderator isn't around... kick is the only tool for hosts to deal with troublesome trolls. Even if moderators are around to deal with a potential situation, they could simply be busy with something else at the time.

**Comment by sprangg** [comment]
maackey, the common case that I've witnessed is that the host isn't a complete prick to everyone else - they just ban some 1-2 persons that they personally don't like from playing in their room, and when this is the open teamgame room that everyone joins, these 1-2 persons are unable to play ZK teamgames at all and the admins are forced to step in.

I don't think anyone who thinks about the good of ZK wants the above scenario, where an arbitrary room boss dictates who gets to play ZK for the day. So it's either the admins who have to prevent this manually, or it could be the game itself preventing this by disabling kick in non-passworded rooms.

**Comment by maackey** [comment]
Agreed, retroactive banning and punishing is not the best solution to the problem. It's still bizarre to me that this would be both a common problem with different players abusing kick rights, and simultaneously not enough players for multiple rooms. One room culture sucks. 

Has anyone yet implemented dynamic room size caps? Under ~12 active players _(players who have actively played in a public room in the last 15 mins or so -- doesn't count passworded rooms or spectators)_ room limits won't really make a big difference, I admit. But during the intermediate times while building up to enough players to fill multiple lobster pots, limiting room sizes will help ensure there's always a game with players to join while another is in progress.

Game availability is the key -- having a singular room where everyone plays limits game availability, particularly in the case here where the host abuses powers. But if there are multiple rooms to join, its a non-issue. I'd even go as far to count spectators as part of the room limit (probably very unpopular opinion) -- being able to join and spectate mid-game is a really neat and great feature but it absolutely contributes to the one-room culture problem.
